internal/balancergroup/balancergroup.go

```diff
@@ -305,11 +281,14 @@ func (bg *BalancerGroup) AddWithClientConn(id, balancerName string, cc balancer.
 	// Store data in static map, and then check to see if bg is started.
 	bg.outgoingMu.Lock()
 	defer bg.outgoingMu.Unlock()
+	if bg.outgoingClosed {
+		return fmt.Errorf("BalancerGroup already closed")
+	}
 	var sbc *subBalancerWrapper
-	// If outgoingStarted is true, search in the cache. Otherwise, cache is
-	// guaranteed to be empty, searching is unnecessary. Also, skip the cache if
-	// caching is disabled.
-	if bg.outgoingStarted && bg.deletedBalancerCache != nil {
+	// If caching is enabled, search in the cache. Otherwise, cache is
+	// guaranteed to be empty, searching is unnecessary. Also, skip the cache
+	// if caching is disabled.
```

Comment:
I think this comment can actually be much simpler.

// Skip searching the cache if disabled.

Reply:
Changed the comment.